app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb

```diff
@@ -109,7 +109,11 @@ def fetch_jwt_claims_to_validate
       def initialize_validate_restrictions
         @restriction_validator = Authentication::AuthnJwt::ValidateRestrictionsOneToOne
         @restrictions_from_annotations_class = Authentication::ResourceRestrictions::GetServiceSpecificRestrictionsFromAnnotation
-        @extract_resource_restrictions = Authentication::ResourceRestrictions::ExtractResourceRestrictions.new(get_restriction_from_annotation: @restrictions_from_annotations_class)
+        @fetch_non_empty_resource_annotations = Authentication::ResourceRestrictions::FetchNonEmptyResourceAnnotations
```

Comment:
should this be `Authentication::ResourceRestrictions::FetchNonEmptyResourceAnnotations.new`?